p2p/peer.go

```diff
@@ -397,6 +397,9 @@ outer:
 func (p *Peer) startProtocols(writeStart <-chan struct{}, writeErr chan<- error) {
 	p.wg.Add(len(p.running))
 	for _, proto := range p.running {
+		if proto == nil {
+			return
```

Comment:
I would continue here, not return, possible there is another protocol that is not nil